### ISSUE TITLE
Technical search syntax exposed in search field

### ISSUE THREAD

**Issue by Valdnet** [nextcloud/server#22984]
### Steps to reproduce
1. Open the Files app.
2. Select Search.
3. Enter the search text that occurs in the Nextcloud instance.
4. Click on the Meatballs menu.

### Expected behaviour
Display only the text strings that should appear.

### Actual behaviour
Displaying text strings such as:

- in:settings_apps
- in:tasks
- in:contacts
- in:files

![obraz](https://user-images.githubusercontent.com/47037905/93745663-2f288c80-fbf4-11ea-9c93-a1e727b1ab84.png)

After applying the filter:

![obraz](https://user-images.githubusercontent.com/47037905/93746590-ac083600-fbf5-11ea-9883-1e57c86c0374.png)

**Nextcloud version:** (see Nextcloud admin page)
20.0.0 beta4

**Comment by kesselb** [comment]
cc @jancborchardt @skjnldsv 

I like the current behavior and also knew it from other services (e.g. GitHub or GitLab).

**Comment by jancborchardt** [comment]
Well, good point that it’s a bit nerdy, but for the first version we are going to go with this, also to partly show people how they can filter stuff.

In the future what we could do is when a filter is enabled, show it as a bubble below the search field with the text "in Tasks" or "in Settings" which you can then x away. Then we don’t have to add that text in the input field anymore.

What do you think @skjnldsv?

**Comment by Spartachetto** [comment]
What if I want to write the syntax down?

Meaning that, being a bit nerdy, I actually write "test in:tasks" to perform the search.

Or, being even less nerdy, I make a first search using the graphical interface and then modify the resulting text.
For example first I look for "test label:Berlin" using the graphical interface and then I modify the string to look for "test label:Hannover" because it is quicker...

I do think that having the **_possibility_** to access the syntax is an added value. Just my two cents...

**Comment by skjnldsv** [comment]
Agree with @kesselb too.
It's also how google or any search provider is doing it.
you don't have to use it, but it's quite straightforward while it looks techincal.

Non tech people will just ignore this and it's fine. Again, same for google, duckduckgo... `site:github.com` `filetype:pdf`... etc etc

Adding a ui for filtering will just make it a lot more complicated and would be a waste of our developing time tbh. 

TL;DR: simple is good, it's working fine, popup is just here for discoverability.
